torchchem/models/multitask_classification.py

```diff
@@ -0,0 +1,126 @@
+"""
+@author: Zhenqin Wu, Bharath Ramsundar
+"""
+
+import torch
+import numpy as np
+from deepchem.metrics import from_one_hot
+from torchchem.models.model import MultitaskModel
+
+
+class MultitaskClassification(MultitaskModel):
+
+  def __init__(self, n_tasks, n_features, n_classes=2, **kwargs):
+    """Constructs the computational graph.
+
+    This function constructs the computational graph for the model. It relies
+    subclassed methods (build/cost) to construct specific graphs.
+
+    Parameters
+    ----------
+    n_tasks: int
+      Number of tasks
+    n_features: int
+      Number of features.
+    n_classes: int
+      Number of classes if this is for classification.
+    """
+    # Save hyperparameters
+    self.n_tasks = n_tasks
+    self.n_features = n_features
+    self.n_classes = n_classes
+    super(MultitaskClassification, self).__init__(**kwargs)
+
+  def build(self):
+    """Constructs the graph architecture as specified in its config.
+
+    This method creates the following Placeholders:
+      mol_features: Molecule descriptor (e.g. fingerprint) tensor with shape
+        batch_size x n_features.
+    """
+
+    layer_sizes = self.layer_sizes
+    weight_init_stddevs = self.weight_init_stddevs
+    bias_init_consts = self.bias_init_consts
+    dropouts = self.dropouts
+    lengths_set = {
+        len(layer_sizes),
+        len(weight_init_stddevs),
+        len(bias_init_consts),
+        len(dropouts),
+    }
+    assert len(lengths_set) == 1, 'All layer params must have same length.'
+    n_layers = lengths_set.pop()
+    assert n_layers > 0, 'Must have some layers defined.'
+
+    prev_layer_size = self.n_features
+    self.W_list = []
+    self.b_list = []
+    for i in range(n_layers):
+      W_init = np.random.normal(0, weight_init_stddevs[i],
+                                (prev_layer_size, layer_sizes[i]))
+      #W_init = torch.cuda.FloatTensor(W_init)
+      W_init = torch.FloatTensor(W_init)

```

Comment:
https://pytorch.org/docs/stable/nn.html#torch.nn.ParameterList

`nn.Parameter` and `nn.ParameterList` are your friends here.

Reply:
As a quick question here, what's best practice in PyTorch models for writing a model that can run on both CPU and GPU? Is there an easy way to do a device type check and run either `W_init = torch.cuda.FlooatTensor` or `W_init = torch.FloatTesnor` depending on GPU or CPU?

Reply:
The scripts I run usually look like this:
```
DEVICE = "cuda" if torch.cuda.is_available() else "cpu"

model = ModelName(**config, device=DEVICE)
model.to(DEVICE)
```

`model.to(DEVICE)` will change all my submodules (`nn.Module`), and `nn.Parameter (nn.ParameterList, nn.ParameterDict)` to cpu/gpu depending on what I have. 

I also use a method on my Dataset or Model class, that allows me to move the input tensors to CPU or GPU.

Reply:
Otherwise, I am not aware of any standard practice. I used this for my thesis and I find it works correctly without any modifications.

Reply:
This is the member function I use for converting input tensors to device
```
def to_device(self, tensors: Union[List, torch.Tensor]) -> Union[List, torch.Tensor]:
        """Converts all inputs to the device used.
        Parameters
        ----------
        tensors: Union[List, torch.Tensor],
            Tensors to convert to model device. The tensors can be either a
            single tensor or an iterable of tensors.
        """
        if isinstance(tensors, list) or isinstance(tensors, tuple):
            tensors = [tensor.to(self.device, non_blocking=True) for tensor in tensors]
            return tensors
        elif isinstance(tensors, torch.Tensor):
            return tensors.to(self.device, non_blocking=True)
        else:
            raise ValueError(f"Tensors of type {type(tensors)} unsupported")
```
This can be a method in our `BaseModel` class and all other DeepChem models can inherit from it.